### PR TITLE
Update UI section, use cases and create group feature implementation in DG

### PR DESCRIPTION
**Pull request by Bearypop**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -808,7 +837,7 @@ testers are expected to do more *exploratory* testing.
 
    1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
 
-   2. Test case: `delete alexy@u.nus.edu`<br>
+   2. Test case: `delete alexyeoh@u.nus.edu`<br>
       Expected: Contact with above email deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
```
Comment (maj0-0) [inline docs/DeveloperGuide.md:841]: Can change the parts that mention contact to student pls

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ The **API** of this component is specified in [`Ui.java`](https://github.com/se-
 
 ![Structure of the UI Component](images/UiClassDiagram.png)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter`, `ConfirmationPopup` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
 
 The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
```
Comment (maj0-0) [inline docs/DeveloperGuide.md:78]: Should we change the links so that its hyperlink to our repo?

**Review by maj0-0 — COMMENTED** [review]
LGTM! Thanks for adding new sequence diagram!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ The **API** of this component is specified in [`Ui.java`](https://github.com/se-
 
 ![Structure of the UI Component](images/UiClassDiagram.png)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter`, `ConfirmationPopup` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
```
Comment (neyapraveen) [inline docs/DeveloperGuide.md:76]: maybe add made up of different parts to add more emphasis

**Review by neyapraveen — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Bearypop) [reply docs/DeveloperGuide.md:841]: Okay

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Bearypop) [reply docs/DeveloperGuide.md:78]: Okay

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Bearypop) [reply docs/DeveloperGuide.md:76]: Okay

**Pull request merged**